src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -27,12 +28,26 @@ internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactor
             IOptions<DurableTaskOptions> options,
             IConnectionStringResolver connectionStringResolver,
             INameResolver nameResolver,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            IPlatformInformationService platformInfo)
         {
             this.options = options.Value;
             this.nameResolver = nameResolver;
             this.loggerFactory = loggerFactory;
             this.azureStorageOptions = new AzureStorageOptions();
+            this.inConsumption = platformInfo.InConsumption();
```

Comment:
I think if we moved this logic down into the GetAzureStorageOrchestrationServiceSettings, it may make certain things cleaner (for instance, we wouldn't need to check for the nullability of this.options.MaxConuccrent*Functions because we know they are explicitly not null.

It is also just nice to have all of the logic regarding settings in one place.

That being said, that may delay the validation logic, so consider this a soft suggestion.

Reply:
I'm a little concerned about delaying the validation step. I do want to follow-up on this suggestion, but I'll defer making a decision on it just yet because I want to test what happens when we move things around. There'll be another opportunity with the OOProc PR :)